tests/components/netatmo/test_config_flow.py

```diff
@@ -108,11 +108,76 @@ async def test_option_flow(hass):
     """Test config flow options."""
     valid_option = {
         "lat_ne": 32.91336,
+        "lon_ne": -117.187429,
+        "lat_sw": 32.83336,
         "lon_sw": -117.26743,
         "show_on_map": False,
         "area_name": "Home",
-        "lon_ne": -117.187429,
-        "lat_sw": 32.83336,
+        "mode": "avg",
+    }
+
+    expected_result = {
+        "lat_ne": 32.9133601,
+        "lon_ne": -117.1874289,
+        "lat_sw": 32.8333601,
+        "lon_sw": -117.26742990000001,
+        "show_on_map": False,
+        "area_name": "Home",
+        "mode": "avg",
+    }
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, unique_id=DOMAIN, data=VALID_CONFIG, options={},
+    )
+    config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "public_weather_areas"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={CONF_NEW_AREA: "Home"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "public_weather"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input=valid_option
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "public_weather_areas"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    for k, v in expected_result.items():
+        assert config_entry.options[CONF_WEATHER_AREAS]["Home"][k] == v
```

Comment:
We should mock out async_setup and async_create_entry so we don't unexpectedly setup and start doing I/O

Reply:
Disregard, I missed that this was an options flow 👍